### PR TITLE
Add datetime.pxd to cython.cpython

### PR DESCRIPTION
**Pull request by intellimath**
In python 2.X/3.X datetime module has it's own datetime.h file in standard (C headers) library. Fast creation of date/time/datetime instances require access to C API of datetime module. Addition of datetime.pxd could resolve this problem.

Given the immutable nature of date/time/datetime/timedelta objects by design we always have to recreate these objects when some of their properties have to change. So addition of inline functions for fast access to their members could help to do that as fast as possible.

### CODE REVIEW

File: Cython/Includes/cpython/datetime.pxd [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    # Getters for time (C macros).
+    int PyDateTime_TIME_GET_HOUR(object o)
+    int PyDateTime_TIME_GET_MINUTE(object o)
+    int PyDateTime_TIME_GET_SECOND(object o)
+    int PyDateTime_TIME_GET_MICROSECOND(object o)
+
+    # Getters for timedelta (C macros).
+    #int PyDateTime_DELTA_GET_DAYS(object o)
+    #int PyDateTime_DELTA_GET_SECONDS(object o)
+    #int PyDateTime_DELTA_GET_MICROSECONDS(object o)
```
Comment (jbrockmendel) [inline Cython/Includes/cpython/datetime.pxd:95]: why are these disabled?

Comment (scoder) [reply Cython/Includes/cpython/datetime.pxd:95]: There are inline functions for the same thing. But, yeah, I don't know. I've uncommented them in b177f2296686833a3903e0b8a7dcf4afbed2e56d.

**Comment by enjoysmath** [comment]
How does one go about installing the cython version that has these datetime interfaces?

**Comment by scoder** [comment]
Yes, this should work. Take the latest `master` branch.

**Pull request merged**